src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1307,6 +1307,7 @@ protected unsafe void WriteEventWithRelatedActivityIdCore(int eventId, Guid* rel
                 Debug.Assert(m_eventData != null);  // You must have initialized this if you enabled the source.
                 try
                 {
+                    m_activeWritesCount++;
```

Comment:
In order for the lockless two volatile system to work the writer thread needs to increase m_activeWritesCount count, then immediately check m_eventSourceEnabled and bail if it is false

Reply:
You're totally right. That's what I get for trying to code too late on a Friday 😉